src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -2689,28 +2702,15 @@ FixedwingPositionControl::tecs_update_pitch_throttle(const hrt_abstime &now, flo
 	_tecs.update_vehicle_state_estimates(_airspeed, _body_acceleration(0), (_local_pos.timestamp > 0), in_air_alt_control,
 					     _current_altitude, _local_pos.vz);
 
-	/* scale throttle cruise by baro pressure */
-	if (_param_fw_thr_alt_scl.get() > FLT_EPSILON) {
-		vehicle_air_data_s air_data;
-
-		if (_vehicle_air_data_sub.copy(&air_data)) {
-			if (PX4_ISFINITE(air_data.baro_pressure_pa) && PX4_ISFINITE(_param_fw_thr_alt_scl.get())) {
-				// scale throttle as a function of sqrt(p0/p) (~ EAS -> TAS at low speeds and altitudes ignoring temperature)
-				const float eas2tas = sqrtf(CONSTANTS_STD_PRESSURE_PA / air_data.baro_pressure_pa);
-				const float scale = constrain((eas2tas - 1.0f) * _param_fw_thr_alt_scl.get() + 1.f, 1.f, 2.f);
-
-				throttle_max = constrain(throttle_max * scale, throttle_min, 1.0f);
-				throttle_cruise = constrain(throttle_cruise * scale, throttle_min + 0.01f, throttle_max - 0.01f);
-			}
-		}
-	}
+	float throttle_trim_min = _param_fw_thr_trim_min.get() >= 0.0f ? _param_fw_thr_trim_min.get() : NAN;
+	float throttle_trim_max = _param_fw_thr_trim_max.get() >= 0.0f ? _param_fw_thr_trim_max.get() : NAN;
```

Comment:
i would either handle this inside TECS (with sanity checks) or make sure to define the inputs to `update_pitch_throttle()` clearly (documentation) describing what NAN is going to mean when input to the method.. otherwise ppl reading it have no idea.

side note: in general.. I'm not the biggest fan of passing NANs around. e.g. (similar to another comment somewhere in this review) this could all be handled in the library and actual floats could be delegated to each value, then the logic in the other methods dont need to constantly check for NAN, and readers dont need to parse what NAN is intended to mean.